core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -523,27 +523,31 @@ protected void startJGroupsChannelIfNeeded() {
          } catch (Exception e) {
             throw new CacheException("Unable to start JGroups Channel", e);
          }
-
-         try {
-            // Normally this would be done by CacheManagerJmxRegistration but
-            // the channel is not started when the cache manager starts but
-            // when first cache starts, so it's safer to do it here.
-            globalStatsEnabled = configuration.statistics();
-            if (globalStatsEnabled) {
-               // TODO Use the overloaded variant when available: https://issues.jboss.org/browse/JGRP-2394
-               JmxConfigurator.registerChannel(channel, jmxRegistration.getMBeanServer(), jmxRegistration.getDomain(), clusterName, true);
-            }
-         } catch (Exception e) {
-            throw new CacheException("Channel connected, but unable to register MBeans", e);
-         }
       }
+      startJmxIfNeeded(clusterName);
       if (!connectChannel) {
          // the channel was already started externally, we need to initialize our member list
          receiveClusterView(channel.getView());
       }
       CLUSTER.localAndPhysicalAddress(clusterName, getAddress(), getPhysicalAddresses());
    }
 
+   // This needs to stay as a separate method to allow for substitution for Substrate
+   private void startJmxIfNeeded(String clusterName) {
```

Comment:
No, this does not start JMX :) Please name it ```registerMBeans```.

Reply:
Okay.